packages/front-end/pages/404.tsx

```diff
@@ -0,0 +1,36 @@
+import * as Sentry from "@sentry/react";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+import { isSentryEnabled } from "@/services/env";
+
+export default function Custom404() {
+  const ind = Math.ceil(Math.random() * 2);
+  const router = useRouter();
+  const badPath = router?.asPath || window?.location?.href || "";
```

Comment:
We cannot normally use `window` and `document` safely in Next.js due to the SSG and SSR features, and in those environments it would throw a ReferenceError. I don't think we are using them so maybe that's not a problem. 

Usually the values from `useRouter()` on first load are empty. Is it possible this executes twice? - once for the `window?.location` evaluation and another for the `router?.asPath` evaluation? `window.location.href` and `router.asPath` I don't think would return the same values--the full URL would be returned from `window.location.href` and only the path from `router.asPath` if I'm not mistaken:

> asPath: String - The path as shown in the browser including the search params and respecting the trailingSlash configuration. basePath and locale are not included.

What I may do in a case like this could be something like this:

```ts
  const badPath = router?.asPath;

  useEffect(() => {
    if (!isSentryEnabled()) return;

    if (!badPath) return;

    if (typeof document === 'undefined') return;

    const referrer = document?.referrer || "";
    Sentry.captureMessage("404: " + badPath + " from " + referrer);
  }, [badPath]);
```

This may mean we also don't get duplicate events for cloud-based errors and self-hosted errors (assuming we use Sentry on self-hosted instances but maybe we don't). 

I haven't run this code so I'm not sure if there's a problem with it.